src/health-score.js

```diff
@@ -2,6 +2,34 @@ const findProblematicComments = require('./score_components/find-problematic-com
 const retrieveCodeCoverage = require('./score_components/coverage');
 const reportStatus = require('./report');
 
+/**
+ *
+ * @param input
+ * @returns []
+ */
+function parseYamlArray(input) {
+  if (!input) {
+    return [];
+  }
+  const arr = input.trim().replace(/^\[|\]$/g, '');
+
+  try {
+    console.log(JSON.parse(arr));
+    return JSON.parse(arr);
+  } catch (e) {
+    return arr
```

Comment:
Can you add a comment as to what is happening here? Do I understand it correctly that this function tries to parse JSON, and if that fails, assumes some YAML syntax is being used (i.e. list) and goes to parse that?

If so, I think it may help to a) extract this function out into its own module and b) to unit test it. I have a feeling there may be edge case bugs lurking in here, would be nice to make sure this is functioning as desired. I think you did a great job covering this functionality in the tests you added to the health score module, I would just factor this out into its own module is all.

Reply:
Yep, it tries parsing JSON and if it fails does regex stuff to parse YAML lists/blocks. This is redundant I believe, as the brackets are being removed before the try catch block. So it basically converts even valid JSON strings to YAML. I believe it can be removed.